src/main/java/seedu/sellsavvy/model/person/Person.java

```diff
@@ -156,4 +156,26 @@ public String toString() {
                 .toString();
     }
 
+    /**
+     * Creates a new copy of this {@code Person} with new attributes.
+     * Only the tags are not copied.
+     */
+    public Person createCopy() {
+        Name name = new Name(getName().toString());
+        Address address = new Address(getAddress().toString());
+        Phone phone = new Phone(getPhone().toString());
+        Email email = new Email(getEmail().toString());
+        Set<Tag> tags = getTags();
+        OrderList orderList = getOrderList().createCopy();
+        Person personCopy = new Person(name, phone, email, address, tags, orderList);
+
+        if (filteredOrders.getPredicate() == null) {
+            return personCopy;
+        }
+
+        @SuppressWarnings("unchecked")
+        Predicate<Order> filterPredicate = (Predicate<Order>) filteredOrders.getPredicate();
```

Comment:
I used assigned to  Predicate<? super Order> for predicate to not use @suppresswarnings, is that better?

Reply:
![image](https://github.com/user-attachments/assets/a069c7f0-a722-48ec-b674-d5414a405708)
should i just change the method signature for `updateFilteredOrderList`?

Reply:
changed